src/wallet/walletutil.cpp

```diff
@@ -25,3 +25,46 @@ fs::path GetWalletDir()
 
     return path;
 }
+
+static bool IsBerkeleyBtree(const fs::path& path)
+{
+    // Berkeley DB Btree magic bytes, from:
+    //  https://github.com/file/file/blob/5824af38469ec1ca9ac3ffd251e7afe9dc11e227/magic/Magdir/database#L74-L75
+    const uint32_t bdb_magic = 0x00053162;
+
+    std::ifstream file(path.string(), std::ios::binary);
+    if (!file.is_open()) return false;
+
+    file.seekg(12, std::ios::beg); // Magic bytes start at offset 12
+    uint32_t data = 0;
+    file.read((char*) &data, sizeof(data)); // Read 4 bytes of file to compare against magic
```

Comment:
Any concerns about the endianness of `data` as compared to `bdb_magic` here?

Reply:
> Any concerns about the endianness of data as compared to bdb_magic here?

Could add a comment here, but this checks for databases in the native format. On big endian systems, the magic bytes are `00 05 31 62`, and on little endian systems they are `62 31 05 00`.

Reply:
agree with regard to adding comment; at least I too had no idea that BerkeleyDB files generated on big-endian systems are incompatible!